Lib/test/test_heapq.py

```diff
@@ -74,13 +74,46 @@ def test_push_pop(self):
         except AttributeError:
             pass
 
+    def test_max_push_pop(self):
```

Comment:
There is a lot of functionality left untested; in particular the `TestErrorHandling` class below doesn't cover max-heaps.
I wonder if these tests should share code, rather than repeat it. Raymond should probably set the general direction here, though.